File: cobaco-server-spring/src/main/java/dentaira/cobaco/server/file/StoredFile.java
package dentaira.cobaco.server.file;

import java.io.InputStream;
import java.nio.file.Path;
import java.util.UUID;

public class StoredFile {

    private final UUID id;

    private String name;

    private Path path;

    private FileType type;

    private InputStream content;

    private DataSize size;

    public StoredFile(UUID id, String name, Path path, FileType type, DataSize size) {
        this.id = id;
        this.name = name;
        this.path = path;
        this.type = type;
        this.size = size;
    }

    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Path getPath() {
        return path;
    }

    public void setPath(Path path) {
        this.path = path;
    }

    public FileType getType() {
        return type;
    }

    public InputStream getContent() {
        return content;
    }

    public void setContent(InputStream content) {
        this.content = content;
    }

    public DataSize getSize() {
        return size;
    }

    public void setSize(DataSize size) {
        this.size = size;
    }

    public String displaySize() {
        return isFile() ? getSize().toString() : "";
    }

    public boolean isFile() {
        return type == FileType.FILE;
    }

    public boolean isDirectory() {
        return type == FileType.DIRECTORY;
    }
}

File: cobaco-server-play/app/dentaira/cobaco/server/file/StoredFile.scala
package dentaira.cobaco.server.file

import scalikejdbc._

import java.io.InputStream
import java.nio.file.{Path, Paths}

case class StoredFile(id: String,
                      name: String,
                      path: Path,
                      `type`: FileType,
                      content: InputStream,
                      size: Long) {
}

object StoredFile extends SQLSyntaxSupport[StoredFile] {

  override val tableName = "file"

  def apply(id: String, name: String, path: Path, `type`: FileType, content: InputStream, size: Long): StoredFile = {
    new StoredFile(id, name, path, `type`, content, size)
  }

  def apply(s: SyntaxProvider[StoredFile])(rs: WrappedResultSet): StoredFile = apply(s.resultName)(rs)

  def apply(s: ResultName[StoredFile])(rs: WrappedResultSet) = new StoredFile(
    id = rs.string(s.id),
    name = rs.string(s.name),
    path = Paths.get(s.path),
    `type` = FileType.valueOf(rs.string(s.`type`)),
    content = rs.binaryStream(s.content),
    size = rs.long(s.size)
  )
}
